src/main/scala-2.12/com/linagora/gatling/imap/action/DisconnectAction.scala

```diff
@@ -0,0 +1,21 @@
+package com.linagora.gatling.imap.action
+
+import akka.actor.Props
+import com.linagora.gatling.imap.protocol.{Command, UserId}
+import io.gatling.core.action.ActionActor
+import io.gatling.core.session.Session
+
+object DisconnectAction {
+  def props(imapContext: ImapActionContext, requestName: String): Props =
+    Props(new DisconnectAction(imapContext, requestName))
+}
+
+class DisconnectAction(val imapContext: ImapActionContext, val requestName: String) extends ActionActor with ImapActionActor {
+
+
+  override def successOnDisconnect: Boolean = true
+
+  override def execute(session: Session): Unit = {
+    sessions.tell(Command.Disconnect(UserId(session.userId)), handleResponse(session, imapContext.clock.nowMillis))
```

Comment:
we should probably change the current actor to an actor expecting the Disconnected event instead of propagating a boolean. WDYT?

Reply:
yes, it is cleaner. But i may revert all of this as i am not sure how to correctly handle the connection/disconnection on an Imap session.
If i reconnect after having disconnected from the session, i have an exception in the imap nio library. 
Do you know if we can reconnect to a previously closed session ?

Reply:
I have no idea

Reply:
I would say no. Maybe you can implement the logout command which is pretty clear about that https://tools.ietf.org/html/rfc3501#section-6.1.3